runtime/util/fltmath.c

```diff
@@ -37,6 +37,28 @@
 #define J9_NO_DENORMAL_FLOAT_SUPPORT 1
 #endif
 
+I_32
+helperCDoubleAreDoublesEqual(jdouble a, jdouble b)
+{
+	I_32 result;
+
+	if (IS_NAN_DBL(a) && IS_NAN_DBL(b)) {
```

Comment:
Do the fast comparison first, `is a = b`. If its not equal then check if both are `NaN`, otherwise return false